swagger/swagger.js

```diff
@@ -48,75 +48,93 @@ module.exports = function(RED) {
+                            for(var i=0; i < swaggerPart.consumes.length; i++){
+                                swaggerPart.consumes[i] = swaggerPart.consumes[i].trim();
+                            }
                         }
-                    }
-                    if(swagger.deprecated){
-                        swaggerPart.deprecated = true;
-                    }
-                    if(swagger.parameters.length > 0){
-                        swaggerPart.parameters = swagger.parameters.slice();
-                        if(additionalParams){
-                            for(var i in additionalParams){
-                                swaggerPart.parameters.push(additionalParams[i]);
+                        if(swagger.produces){
+                            swaggerPart.produces = swagger.produces.split(',');
+                            for(var i=0; i < swaggerPart.produces.length; i++){
+                                swaggerPart.produces[i] = swaggerPart.produces[i].trim();
                             }
                         }
-                    } else if(additionalParams){
-                        swaggerPart.parameters = additionalParams.slice();
-                    }
-                    if(Object.keys(swagger.responses).length > 0){
-                        swaggerPart.responses = swagger.responses;
+                        if(swagger.deprecated){
+                            swaggerPart.deprecated = true;
+                        }
+                        if(swagger.parameters.length > 0){
+                            swaggerPart.parameters = swagger.parameters.slice();
+                            if(additionalParams){
+                                for(var i in additionalParams){
+                                    swaggerPart.parameters.push(additionalParams[i]);
+                                }
+                            }
+                        } else if(additionalParams){
+                            swaggerPart.parameters = additionalParams.slice();
+                        }
+                        if(Object.keys(swagger.responses).length > 0){
+                            swaggerPart.responses = swagger.responses;
+                        } else{
+                            swaggerPart.responses = {
+                                default: {}
+                            };
+                        }
+                        node.status({});
                     } else{
+                        swaggerPart.summary = node.name || (node.method+" "+url);
                         swaggerPart.responses = {
                             default: {}
                         };
+                        if(additionalParams){
+                            swaggerPart.parameters = additionalParams.slice();
+                        }
+                        node.status({fill:"yellow",shape:"ring",text:RED._("swagger.status.missingconfig")});
```

Comment:
Yep. Will play with it some more in a bit and update the pull request.
On Aug 4, 2015 16:07, "knolleary" <notifications@github.com> wrote:

> In swagger/swagger.js
> <https://github.com/node-red/node-red-node-swagger/pull/19#discussion_r36242079>
> :
>
> >                          swaggerPart.responses = {
> >                              default: {}
> >                          };
> > +                        if(additionalParams){
> > +                            swaggerPart.parameters = additionalParams.slice();
> > +                        }
> > +                        node.status({fill:"yellow",shape:"ring",text:RED._("swagger.status.missingconfig")});
>
> It ought to Just Work - had you reloaded the editor to ensure it had
> picked up the messages in the catalog?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/node-red/node-red-node-swagger/pull/19/files#r36242079>
> .
>